Objects/bytesobject.c

```diff
@@ -2181,6 +2181,68 @@ bytes_replace_impl(PyBytesObject *self, Py_buffer *old, Py_buffer *new,
 
 /** End DALKE **/
 
+/*[clinic input]
+bytes.cutprefix as bytes_cutprefix
+
+    prefix: Py_buffer
+    /
+
+Remove a specified prefix, if present.
+
+If the bytes starts with the prefix, return b[len(prefix):].
+Otherwise, return the original bytes.
+[clinic start generated code]*/
+
+static PyObject *
+bytes_cutprefix_impl(PyBytesObject *self, Py_buffer *prefix)
+/*[clinic end generated code: output=5e5ef0cf576a8a9c input=f5409d4eb86164e8]*/
+{
+    const char *self_start = PyBytes_AS_STRING(self);
+    Py_ssize_t self_len = PyBytes_GET_SIZE(self);
+    const char *prefix_start = prefix->buf;
+    Py_ssize_t prefix_len = prefix->len;
+
+    if (self_len >= prefix_len
+        && memcmp(self_start, prefix_start, prefix_len) == 0)
+    {
+        return PyBytes_FromStringAndSize(self_start + prefix_len,
+                                         self_len - prefix_len);
+    }
+
+    return PyBytes_FromStringAndSize(self_start, self_len);
```

Comment:
May be add a special case when return `self` unchanged (`PyBytes_CheckExact(self)`)?